eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java

```diff
@@ -3,18 +3,58 @@
 
 package com.azure.eventhubs;
 
+import com.azure.core.amqp.AmqpConnection;
+import com.azure.core.exception.AzureException;
+import com.azure.eventhubs.implementation.ReactorConnection;
+import com.azure.eventhubs.implementation.ReactorHandlerProvider;
+import com.azure.eventhubs.implementation.ReactorProvider;
+import com.azure.eventhubs.implementation.StringUtil;
+import com.azure.eventhubs.implementation.TokenProvider;
 import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
 
-// Each time a build method is called, a new receiver or sender is created.
-class EventHubClient implements AutoCloseable {
+import java.io.Closeable;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.Period;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
 
-    EventHubClient() {
+/**
+ * The main point of interaction with Azure Event Hubs, the client offers a connection to a specific Event Hub within
+ * the Event Hubs namespace and offers operations for sending event data, receiving events, and inspecting the connected
+ * Event Hub.
+ */
+public class EventHubClient implements Closeable {
+    private final String connectionId;
+    private final Mono<AmqpConnection> connectionMono;
+    private final String host;
+    private final AtomicBoolean hasConnection = new AtomicBoolean(false);
+
+    //TODO (conniey): Can we remove this and replace with an TokenProvider?
+    private final ConnectionStringBuilder connectionStringBuilder;
+    //TODO (conniey): Replace with configured values in EventHubClientBuilder.
+    private final Duration timeout = Duration.ofSeconds(45);
+
+    EventHubClient(ConnectionStringBuilder connectionStringBuilder, TokenProvider tokenProvider,
+                   ReactorProvider provider, ReactorHandlerProvider handlerProvider, Scheduler scheduler) {
+        Objects.requireNonNull(connectionStringBuilder, "'connectionStringBuilder' is null");
+        Objects.requireNonNull(connectionStringBuilder.endpoint(), "'connectionStringBuilder.endpoint()' is null.");
+
+        this.connectionStringBuilder = connectionStringBuilder;
+        this.host = connectionStringBuilder.endpoint().getHost();
+        this.connectionId = StringUtil.getRandomString("MF");
+        this.connectionMono = Mono.fromCallable(() -> ReactorConnection.create(connectionId, host, tokenProvider, provider, handlerProvider, scheduler))
+            .doOnSubscribe(c -> hasConnection.set(true))
+            .cache();
     }
 
     /**
      * Creates a builder that can configure options for the {@link EventHubClient} before creating an instance of it.
      *
-     * @return A new {@link EventHubClientBuilder} to create an EventHubClient from.
+     * @return A new {@link EventHubClientBuilder} to createReactor an EventHubClient from.
```

Comment:
find & replace fail

Reply:
I blame IntelliJ for trying to replace in docs, too. 🤣 I need to turn off that feature.